src/components/Table/GridTable.tsx

```diff
@@ -284,11 +292,26 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
     return rows;
   }, [columns, rows, sorting, sortState]);
 
-  const columnSizes = useMemo(
-    () => calcColumnSizes(columns, style.nestedCards?.firstLastColumnWidth),
-    [columns, style.nestedCards?.firstLastColumnWidth],
+  const [columnSizes, setColumnSizes] = useState<string[]>(
+    calcColumnSizes(columns, style.nestedCards?.firstLastColumnWidth, undefined, style.minWidth),
   );
 
+  const [tableWidth, setTableWidth] = useState<number | undefined>();
+
+  const debouncedTableWidth = useDebouncedCallback((width) => {
```

Comment:
Maybe call this `setTableWidthDebounced`? Pretty neat, I just didn't get it was a setter.